### DISCUSSION TITLE
NotFound routing to 404 in production?

### DISCUSSION THREAD

**Discussion by VmMad** [facebook/docusaurus#8881] [Q&A]
I'm developing an app with with 3 different versions.
Most of them have the same documentation but served differently for each version. 
I'm trying to make a switcher to be able to change the version of the documentation.
This switcher makes use of the `useLocation` hook to get the pathname so I can make different dynamic routes. The thing is, sometimes the documentation doesn't exists for some versions. For example, let's say I have 3 base urls: `/water`, `/fire`, and `/air`. If I go to `/water/burn-this` I get the NotFound component. This is right because I don't have this  documentation yet for water, but I do have it in `/fire` so the switcher should get me to `/fire/burn-this` but instead it changes the links to `/fire/404` and `/air/404` even though the current pathname is `/burn-this`.

I was wondering if this is a docusaurus thing treating the NotFound routes as 404 under the hood, and just applies this change when building the app, because locally it works.

**Comment by slorber** [comment]
It's hard to understand your use-case, a repro would help using docusaurus.new starting from an empty project.

I am not aware of redirections to `/xyz/404`. It's likely your production host that does that redirect. 
We usually serve 404 pages without redirection, see https://docusaurus.io/docs/doesNotExist

**Reply by VmMad** [reply]
https://docusaurus.io/docs/doesNotExist is basically an example. When you get to the NotFound and then try to change the language you will be redirected to `https://docusaurus.io/<lang>/404`. This is exacly what happens to me. I created a switcher that uses the `useLocation` hook to get the pathname but in production this always redirect to 404 instead of the actual pathname
